data/components/frame-list.js

```diff
@@ -182,9 +183,25 @@ var FrameBubble = React.createFactory(React.createClass({
     }
 
     if (!preview && frame.mqtt) {
+      var mqtt = frame.mqtt;
+      var data;
+      op = "MQTT " + mqtt.cmd;
+      if (mqtt.cmd === "publish") {
+        payload = mqtt.topic;
+        data = mqtt.payload;
+      } else {
+        payload = mqtt.messageId || mqtt.clientId || mqtt.returnCode;
+        data = {packet: mqtt};
+        if (mqtt.subscriptions)
+          data = {subscriptions: mqtt.subscriptions};
+        if (mqtt.granted)
+          data = {granted: mqtt.granted};
+        if (mqtt.unsubscriptions)
+          data = {unsubscriptions: mqtt.unsubscriptions};
+      }
       preview = TreeView({
         key: "preview-mqtt",
-        data: {"MQTT": frame.mqtt},
+        data: data,
```

Comment:
Do you mind writing this as `data: {"MQTT": data}`? That way we keep the view consistent with the other protocols.